### PR TITLE
Publishing system and reply model

### PR DESCRIPTION
**Pull request by TheJhyde**
Let's you publish tweets to twitter, edit posted tweets, and revises the reply system so it points you to preposted chains.

### CODE REVIEW

**Comment by TheJhyde** [comment]
I looked over the conflicts, they're easily fixed. I can fix them once this is approved for merging.

**Comment by vkoves** [comment]
Since @TheJhyde merged this accidentally, I'll be going through and reviewing this.

File: app/controllers/lessons_controller.rb [hunk 1]
```diff
@@ -89,13 +89,18 @@ def update
...
 
+  def publish
+    @lesson = Lesson.find(params[:id])
+    @lesson.publish
+ 
+    flash["info"] = "This lesson has been published to Twitter."
+    redirect_to lesson_url(@lesson.id)
+  end
+
   def destroy
```
Comment (vkoves) [inline app/controllers/lessons_controller.rb]: Also not in your work, but does controller action make any sense? I'm not sure why the instance variable would be present frankly, it should be running `@lesson.destroy` after doing a find. Also `destroy` is preferred over delete as it runs any callback and clean up (like deleting tweets if we have that implemented in the model), while `.delete does not`.

File: app/models/mention.rb [hunk 2]
```diff
@@ -53,7 +53,7 @@ def self.reply_mentions
       if @lesson
         twitters = Twitter_API.new
         @client = twitters.client
-        to_send = mention.handler + ' ' + @lesson.intro + @lesson.thread_link
+        to_send = "#{mention.handler} #{@lesson.intro} #{@lesson.thread_link}"
```
Comment (TheJhyde) [inline app/models/mention.rb]: reply_mentions tweets the response to a mention, so it links to a threaded lesson but it tweets everytime the bot is called into action. I think that functionality makes more sense in the mention model.

File: app/controllers/lessons_controller.rb (hunk 1 above)
Comment (TheJhyde) [inline app/controllers/lessons_controller.rb]: @lesson is defined by the find_lesson method that gets run before_action up at the top. I can switch to the destroy.

**Pull request merged**